### ISSUE TITLE
2.9.2 Object self reference crashes swagger-ui

### ISSUE THREAD

**Issue by lmeisegeier** [swagger-api/swagger-ui#4707]
Since Version 2.9.2 swagger-ui crashes If one tries to open controller endpoint with an object that contains a self reference.
![firefoxswaggeruicrash](https://user-images.githubusercontent.com/40823834/42277658-6a730124-7f98-11e8-91ce-46a2ea709598.PNG)

Object that contains cyclic self reference: (ommited other attributes):
```
@Entity
@Table(name = "roles")
public class Role {

...

  @ManyToOne
  @JoinColumn(name = "ancestor_id")
  @JsonProperty
  private Role ancestor;

...

  public Role getAncestor() {
    return this.ancestor;
  }

  public void setAncestor(final Role ancestor) {
    this.ancestor = ancestor;
  }

...

}
```
Firefox show error:  "[Invalid Object]":
![firefoxinvalidobject](https://user-images.githubusercontent.com/40823834/42277657-6a5b4980-7f98-11e8-8dc4-71eff6397572.PNG)

Chrome shows error: "RangeError: Maximum call stack size exceeded":
![chromerangeerror](https://user-images.githubusercontent.com/40823834/42277656-6a4161b4-7f98-11e8-8295-6b55a9b4159d.PNG)

We used version 2.8.0 before and didnt get this issue. We want to use version 2.9.2 because support for Pageable has been added. Our current Workaround is using version 2.8.0 with ApiImplicitParams annotations.
 
Maybe this is a duplicate to https://github.com/swagger-api/swagger-ui/issues/3325 but I didn't read anything about an Invalid Object or RangeError.

**Comment by shockey** [comment]
@lmeisegeier, thanks for filing an issue. There are a couple of problems with your report:

1. 2.8.0 and 2.9.2 aren't valid Swagger UI versions - and from your screenshot, it looks like you're using v3... so you're probably quoting a version of a downstream project that has a different versioning scheme. Please use [our version detection guide](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/version-detection.md) to figure out what Swagger UI versions you are/aren't seeing this with.
2. We can't use annotations to reproduce problems, so it'd help if you could provide the Swagger/OpenAPI document that results from the annotations you posted above 😄

**Comment by SeaLife** [comment]
Hey,

we got the same problem with this since 3.12.x

We've rolled back to 3.3.2 which is working fine so far...

Encountered the issue in 3.12.x, 3.20.2

So, i cant share our code, but its simply a model definition with a array-subitem reference on the same item.

![bildschirmfoto 2018-12-13 um 13 47 56](https://user-images.githubusercontent.com/6567620/49939694-cf109c80-fedd-11e8-9f05-006f5e4350af.png)

Regards
SeaLife (1&1, Germany)

**Comment by shockey** [comment]
Closing due to lack of response from OP.

This is simply to keep our issue tracker clean - feel free to comment if there are any further thoughts or concerns, and we'll be happy to reopen this issue.

@SeaLife, if you'd like to have your issue addressed, please open a new ticket! We'll need more information though - an example of a model definition with an array-subitem reference that causes trouble will suffice if you can't share your real definition.

**Comment by SeaLife** [comment]
Just to mention it, the Bug is solved under version 3.21.0 :)
Thanks @shockey
